multinode_integration_tests/tests/verify_bill_payment.rs

```diff
@@ -24,20 +31,672 @@ use node_lib::database::db_initializer::{
+    let owed_to_serving_node_3_minor =
+        gwei_to_wei::<u128, _>(payment_thresholds.debt_threshold_gwei) + 789_012;
+    let cons_node_initial_service_fee_balance_minor =
+        gwei_to_wei::<u128, _>(payment_thresholds.debt_threshold_gwei) * 4;
+    let exp_final_cons_node_service_fee_balance_minor = cons_node_initial_service_fee_balance_minor
+        - (owed_to_serving_node_1_minor
+            + owed_to_serving_node_2_minor
+            + owed_to_serving_node_3_minor);
+    let test_global_config = TestInputsOutputsConfig {
+        ui_ports_opt: None,
+        cons_node_initial_transaction_fee_balance_minor_opt: None,
+        cons_node_initial_service_fee_balance_minor,
+        debts_config: Either::Left(SimpleSimulatedDebts {
+            owed_to_serving_node_1_minor,
+            owed_to_serving_node_2_minor,
+            owed_to_serving_node_3_minor,
+        }),
+        payment_thresholds_all_nodes: payment_thresholds,
+        cons_node_transaction_fee_agreed_unit_price_opt: None,
+        exp_final_cons_node_transaction_fee_balance_minor: 999_842_470_000_000_000,
+        exp_final_cons_node_service_fee_balance_minor,
+        exp_final_service_fee_balance_serv_node_1_minor: owed_to_serving_node_1_minor,
+        exp_final_service_fee_balance_serv_node_2_minor: owed_to_serving_node_2_minor,
+        exp_final_service_fee_balance_serv_node_3_minor: owed_to_serving_node_3_minor,
+    };
+
+    let stimulate_payments =
+        |cluster: &mut MASQNodeCluster,
+         real_consuming_node: &MASQRealNode,
+         _global_test_config: &TestInputsOutputsConfig| {
+            for _ in 0..6 {
+                cluster.start_real_node(
+                    NodeStartupConfigBuilder::standard()
+                        .chain(Chain::Dev)
+                        .neighbor(real_consuming_node.node_reference())
+                        .build(),
+                );
+            }
+        };
+
+    let start_serving_nodes_and_run_check =
+        |cluster: &mut MASQNodeCluster,
+         serving_node_1_attributes: ServingNodeAttributes,
+         serving_node_2_attributes: ServingNodeAttributes,
+         serving_node_3_attributes: ServingNodeAttributes,
+         _test_global_config: &TestInputsOutputsConfig| {
+            let serving_node_1 = cluster.start_named_real_node(
+                &serving_node_1_attributes.name,
+                serving_node_1_attributes.index,
+                serving_node_1_attributes.config,
+            );
+            let serving_node_2 = cluster.start_named_real_node(
+                &serving_node_2_attributes.name,
+                serving_node_2_attributes.index,
+                serving_node_2_attributes.config,
+            );
+            let serving_node_3 = cluster.start_named_real_node(
+                &serving_node_3_attributes.name,
+                serving_node_3_attributes.index,
+                serving_node_3_attributes.config,
+            );
+            for _ in 0..6 {
```

Comment:
You and I see a possibility of optimisation here:

1) Total 9 Nodes were created, maybe we can get by with a lower number (3 + 3).
2) I'm unsure whether it's the necessity of the test to provide three Node references to each extra Node (the one in loop) that you want to start.